### PR TITLE
Basic build GitHub actions (implementation of hifi PR 16401)

### PR DESCRIPTION
**Pull request by thoys**
This PR enabled GHA (GitHub Actions) to build on each submitted PR based. For this first PR only windows builds are enabled the android/quest, OSX and Linux builds will be enabled in succeeding PR's. The PR is based on High Fidelities GHA commits that were merged into their master branch.

During the building of PR's I ran into GHA out of space issues, which is why I chose to implement a  way to make a release or debug only build for vcpkg. This should for now solve the out of disk space issue. This option is accible through a define in cmake, you could set `VCPKG_BUILD_TYPE` to either `release` or `debug`.  By not setting it, it will build both build types like normally. 
Example to build as vcpkg as release in your cmake generate command:
`cmake .. -DVCPKG_BUILD_TYPE=release`

### CODE REVIEW

**Comment by athena-build-notifier** [comment]
Build artifact uploaded as [HighFidelity-Beta-Interface-PR52-27df651.dmg](https://athena-public.s3.amazonaws.com/pr-builds/kasenvr/project-athena/HighFidelity-Beta-Interface-PR52-27df651.dmg)

**Comment by athena-build-notifier** [comment]
Build artifact uploaded as [HighFidelity-Beta-PR52-27df651.dmg](https://athena-public.s3.amazonaws.com/pr-builds/kasenvr/project-athena/HighFidelity-Beta-PR52-27df651.dmg)

**Comment by athena-build-notifier** [comment]
Build artifact uploaded as [HighFidelity-Beta-Interface-PR52-27df651.exe](https://athena-public.s3.amazonaws.com/pr-builds/kasenvr/project-athena/HighFidelity-Beta-Interface-PR52-27df651.exe)

**Comment by athena-build-notifier** [comment]
Build artifact uploaded as [HighFidelity-Beta-PR52-27df651.exe](https://athena-public.s3.amazonaws.com/pr-builds/kasenvr/project-athena/HighFidelity-Beta-PR52-27df651.exe)

**Comment by athena-build-notifier** [comment]
The following links are available: 
**build_full_win32**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR52/ProjectAthena-Alpha-PR52-6781113_e0ba4973.exe

File: prebuild.py [hunk 1]
```diff
@@ -112,16 +115,39 @@ def main():
...
+    # If not android, install our Qt build
+    if not args.android:
+        qt = hifi_qt.QtDownloader(args)
+        qtInstallPath = qt.cmakePath
+        with hifi_singleton.Singleton(qt.lockFile) as lock:
+            with timer('Qt'):
+                qt.installQt()
+                qt.writeConfig()
+
     # Only allow one instance of the program to run at a time
```
Comment (ctrlaltdavid) [inline prebuild.py:143]: Move this comment down to just above `with hifi_singleton…`.

**Comment by MarcusLlewellyn** [comment]
Unofficial PR build for Windows available here: [ProjectAthena-Alpha-PR52-e0ba497.exe](https://cdn.vircadia.com/dist/pr-builds/ProjectAthena-Alpha-PR52-e0ba497.exe)

File: prebuild.py (hunk 1 above)
Comment (thoys) [reply prebuild.py:143]: ah, good catch, not sure how it got here

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build_full_win32**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR52/ProjectAthena-Alpha-PR52-9ef15bc_f75a0dc9.exe

**Comment by ctrlaltdavid** [comment]
I tested merging this PR into the putative new "master" branch prepresented by https://github.com/kasenvr/project-athena/pull/285 .

There are 26 files chaged.
- 10 without merge conflicts.
- 16 with merge conflicts.

The new "master" branch has GitHub Actions for both PRs and merges into "master", whereas this PR has GitHub Actions only for PRs.

**Comment by thoys** [comment]
> I tested merging this PR into the putative new "master" branch prepresented by #285 .
> 
> There are 26 files chaged.
> 
> * 10 without merge conflicts.
> * 16 with merge conflicts.
> 
There should probably be conflicts in port cmake files (because of the debug/release separation) and some of the python scripts.

> The new "master" branch has GitHub Actions for both PRs and merges into "master", whereas this PR has GitHub Actions only for PRs.

Yes, but also, the latest master doesn't make PR builds downloadable.


What do you suggest I do with this PR? Rebase your PR on top of this and fix the issues?

**Comment by ctrlaltdavid** [comment]
My PR is currently stuck on the Ubuntu problem. Assuming that the problem is solved and that PR merged into the "master-merge" branch, I wonder if the new bits from this PR (e.g., working around GHA build limits) could be applied in a new PR on the "master-merge" branch?

**Comment by thoys** [comment]
Closing this PR in favor of https://github.com/kasenvr/project-athena/pull/300

**Pull request closed without merging**